bindings/java/src/org/sleuthkit/datamodel/OsAccountRealmManager.java

```diff
@@ -0,0 +1,400 @@
+				+ " WHERE LOWER(realms.name) = LOWER('" + realmName + "')"
+				+ " AND " + whereHostClause 
+				+ " ORDER BY realms.host_id IS NOT NULL, realms.host_id";	// ensure that non null host_id are at the front
+				
+		try (Statement s = connection.createStatement();
+				ResultSet rs = connection.executeQuery(s, queryString)) {
+			
+			OsAccountRealm accountRealm = null;
+			if (rs.next()) {
+				Host realmHost = null;
+				long hostId = rs.getLong("host_id");
+				if (!rs.wasNull()) {
+					if (host != null ) {
+						realmHost = host;
+					} else {
+						realmHost = new Host(hostId, rs.getString("host_name"));
+					}
+				}
+				
+				accountRealm = new OsAccountRealm(rs.getLong("realm_id"), rs.getString("realm_name"),
+												RealmNameType.fromID(rs.getInt("name_type")),
+												rs.getString("realm_addr"), realmHost);
+				
+			} 
+			return accountRealm;
+		} catch (SQLException ex) {
+			throw new TskCoreException(String.format("Error getting account realm for with name = %s", realmName), ex);
+		}
+	}
+	
+
+	/**
+	 * Creates a OsAccountRealm from the resultset of a REALM_QUERY_STRING query.
+	 * 
+	 * @param rs ResultSet
+	 * @return
+	 * @throws SQLException 
+	 */
+	private OsAccountRealm resultSetToAccountRealm(ResultSet rs) throws SQLException {
+		
+		long hostId = rs.getLong("host_id");
+		Host realmhost = null;
+		if (!rs.wasNull()) {
+			realmhost = new Host(hostId, rs.getString("host_name"));
+		}
+
+		return new OsAccountRealm(rs.getLong("realm_id"), rs.getString("realm_name"),
+				RealmNameType.fromID(rs.getInt("name_type")),
+				rs.getString("realm_addr"), realmhost);
+	}
+	
+	/**
+	 * Get all realms.
+	 * 
+	 * @return Collection of OsAccountRealm
+	 */
+//	Collection<OsAccountRealm> getRealms() throws TskCoreException {
+//		String queryString = "SELECT realms.id as realm_id, realms.name as realm_name, realms.realm_addr as realm_addr, realms.host_id, realms.name_type, "
+//				+ " hosts.id, hosts.name as host_name "
+//				+ " FROM tsk_os_account_realms as realms"
+//				+ "		LEFT JOIN tsk_hosts as hosts"
+//				+ " ON realms.host_id = hosts.id";
+//
+//		try (CaseDbConnection connection = this.db.getConnection();
+//				Statement s = connection.createStatement();
+//				ResultSet rs = connection.executeQuery(s, queryString)) {
+//
+//			ArrayList<OsAccountRealm> accountRealms = new ArrayList<>();
+//			while (rs.next()) {
+//				long hostId = rs.getLong("host_id");
+//				Host host = null;
+//				if (!rs.wasNull()) {
+//					host = new Host(hostId, rs.getString("host_name"));
+//				}
+//
+//				accountRealms.add(new OsAccountRealm(rs.getLong("realm_id"), rs.getString("realm_name"),
+//						RealmNameType.fromID(rs.getInt("name_type")),
+//						rs.getString("realm_addr"), host));
+//			}
+//
+//			return accountRealms;
+//		} catch (SQLException ex) {
+//			throw new TskCoreException(String.format("Error running the realms query = %s", queryString), ex);
+//		}
+//	}
```

Comment:
Do we need this commented out code?

Reply:
I'm going to leave this in here at this time.   This represent  the category of methods labeled in the document as Maybe/Later/If Needed. 

We can make a final pass at all the API files once all the dust is settled and deleted comment code we know for sure isn't needed.